Watchman.Integrations/Database/MongoDB/MongoConfiguration.cs

```diff
@@ -25,14 +32,53 @@ private static void RegisterConventions()
             ConventionRegistry.Register("DevscordConventions", new MongoConventions(), x => true);
         }
 
-        private class MongoConventions : IConventionPack
+        private static void TryToMigrateDatabase(IConfigurationRoot configuration)
         {
-            public IEnumerable<IConvention> Conventions => new List<IConvention>
+            var connectionString = configuration.GetConnectionString("Mongo");
+            var databaseName = connectionString.Split('/').Last();
+            var assemblyForMigrations = Assembly.GetExecutingAssembly();
+
+            if (ShouldNotDoMigrations(connectionString, databaseName, assemblyForMigrations))
+            {
+                return;
+            }
+
+            var result = new MigrationEngine()
+                .UseDatabase(connectionString, databaseName)
+                .UseAssembly(assemblyForMigrations)
+                .UseSchemeValidation(enabled: false)
+                .UseProgressHandler(context =>
+                    Log.Information($@"The migration with name ""{context.MigrationName}"" from {context.TargetVersion} version was processed as {context.CurrentNumber} from all {context.TotalCount} migrations."))
+                .Run();
+
+            if (!result.Success)
             {
-                new IgnoreExtraElementsConvention(true),
-                new EnumRepresentationConvention(BsonType.String),
-                new CamelCaseElementNameConvention()
-            };
+                // TODO: Do something when migrations throw exceptions.
```

Comment:
Właśnie co zrobić gdy wystąpi wyjątek podczas migracji? Zakończyć aplikacje z odpowiednią informacją?

Poza tym swoją drogą jak teraz patrze w dokumentacje tej paczki to chyba będę musiał użyć try catch aby to obsłużyć zamiast tego if'a.